### PR TITLE
[SPARK-46188][Doc] Fix the CSS of Spark doc's generated tables

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

After https://github.com/apache/spark/pull/40269, there is no border in the generated tables of Spark doc(for example,  https://spark.apache.org/docs/latest/sql-ref-ansi-compliance.html) .  This PR is to fix it by restoring part of the table style in https://github.com/apache/spark/pull/40269/files#diff-309b964023ca899c9505205f36d3f4d5b36a6487e5c9b2e242204ee06bbc9ce9L26

This PR also unifies all the styles of tables by removing the `class="table table-striped"` in HTML style tables in markdown docs.

### Why are the changes needed?

Fix a regression in the table CSS of Spark docs

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Manually build docs and verify.
Before changes:
image

After changes:
image

### Was this patch authored or co-authored using generative AI tooling?

Generated-by: ChatGPT 4

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @grundprinzip as well

File: docs/css/custom.css [hunk 1]
```diff
@@ -1111,5 +1111,18 @@ img {
 table {
   width: 100%;
   overflow-wrap: normal;
+  border-collapse: collapse; /* Ensures that the borders collapse into a single border */
 }
 
+table th, table td {
+  border: 1px solid #cccccc; /* Adds a border to each table header and data cell */
```
Comment (gengliangwang) [inline docs/css/custom.css:1118]: I asked ChatGPT to recover the CSS with doc. It looks good so I keep it.

Comment (gengliangwang) [reply docs/css/custom.css:1118]: The result of the changes are done by myself.

Comment (grundprinzip) [reply docs/css/custom.css:1118]: why dropping the striped this is exactly what the bootstrap theme already provides us. 

The easier way would be simply to 

```
.table {
  border-collapse: collapse;
  border: 1px solid #ccc
}
```

Comment (gengliangwang) [reply docs/css/custom.css:1118]: I tried the striped style but it was not very clear when there is long text ( you can find it in the commit history):
image

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @gengliangwang .

**Review by grundprinzip — APPROVED** [review]

**Review by viirya — APPROVED** [review]

**Comment by gengliangwang** [comment]
Created backport for 3.5 in https://github.com/apache/spark/pull/44097

**Pull request closed without merging**